### PR TITLE
Add book dto mapstruct book controller exception

### PR DESCRIPTION
**Pull request by andrey-kiyas**
Added:
- BookDto
- Mapstruct
- BookController
- Exception

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -12,20 +14,33 @@
...
+<!--        <lombok.mapstruct.binding.version>0.1.0</lombok.mapstruct.binding.version>-->
+<!--        <mapstract.version>1.5.5.Final</mapstract.version>-->
+<!--    </properties>-->
+
+
+        <properties>
+            <java.version>17</java.version>
+            <jdk.version>17</jdk.version>
+            <maven.compiler.source>17</maven.compiler.source>
+            <maven.compiler.target>17</maven.compiler.target>
```
Comment (Rommelua) [inline pom.xml]: why do you need 4 properties with a value "17"?

File: src/main/java/com/onlinebookstore/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @GetMapping("/{id}")
+    public BookDto findById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    @GetMapping("/by-title")
+    public List<BookDto> getAllByTitle(@RequestParam String title) {
+        return bookService.getAllByTitle(title);
+    }
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/controller/BookController.java]:
```suggestion
```

File: src/main/java/com/onlinebookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.onlinebookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+    private Long id;
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
```

File: src/main/java/com/onlinebookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+//@Data
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/model/Book.java]:
```suggestion
```

File: src/main/java/com/onlinebookstore/repository/BookRepository.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.Optional;
+
+public interface BookRepository {
+    Book save(Book book);
+
+    Optional<Book> findById(Long id);
+
+    List<Book> findAll();
+
+    List<Book> findAllByTitle(String title);
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/repository/BookRepository.java]:
```suggestion
```

File: src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            Book book = entityManager.find(Book.class, id);
+            //return (book != null) ? Optional.of(book) : Optional.empty();
+            return Optional.ofNullable(book);
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Can't find books by id: " + id);
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java]: DataProcessingException

File: src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            return entityManager.createQuery("SELECT b FROM Book b", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Can't find all books");
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java]: DataProcessingException

File: src/main/java/com/onlinebookstore/service/BookService.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.List;
+
+public interface BookService {
+    BookDto save(CreateBookRequestDto requestDto);
+
+    List<BookDto> findAll();
+
+    BookDto findById(Long id);
+
+    List<BookDto> getAllByTitle(String title);
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/service/BookService.java]:
```suggestion
```

File: src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        //Optional<Book> optionalBook = bookRepository.findById(id);
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java]: delete commented code

File: src/main/resources/init_db.sql [hunk 10]
```diff
@@ -0,0 +1 @@
+CREATE SCHEMA `online_book_store` DEFAULT CHARACTER SET utf8;
```
Comment (Rommelua) [inline src/main/resources/init_db.sql]: delete this file

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/resources/init_db.sql (hunk 10 above)
Comment (andrey-kiyas) [reply src/main/resources/init_db.sql]: Ok

File: src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java (hunk 9 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java]: Ok

File: src/main/java/com/onlinebookstore/service/BookService.java (hunk 8 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/service/BookService.java]: Ok

File: src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java (hunk 7 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java]: Ok

File: src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java (hunk 6 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java]: Ok

File: src/main/java/com/onlinebookstore/repository/BookRepository.java (hunk 5 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/repository/BookRepository.java]: Ok

File: src/main/java/com/onlinebookstore/model/Book.java (hunk 4 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/model/Book.java]: Ok

File: src/main/java/com/onlinebookstore/dto/CreateBookRequestDto.java (hunk 3 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/dto/CreateBookRequestDto.java]: Ok

File: src/main/java/com/onlinebookstore/controller/BookController.java (hunk 2 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/controller/BookController.java]: Ok

File: src/main/java/com/onlinebookstore/model/Book.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "books")
+public class Book extends BookDto {
```
Comment (sokimaaa) [inline src/main/java/com/onlinebookstore/model/Book.java]: bad idea to extend `BookDto` here.

1. Classes `Book` and `BookDto` have different purposes.
2. Didn't use the inheritance

Leave just 
```
public class Book {
```

Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/model/Book.java]: Yes, strange idea. I don't understand how I wrote this. Already corrected. Thank you!

**Pull request merged**